es6-shim.js

```diff
@@ -291,6 +291,13 @@
       return x != null && Object(x) === x;
     },
 
+    TypeIsNotObject: function (val) {
+      // This is faster when TypeIsObject may commonly return false.
+      return val === void 0 || val === null || val === true || val === false ||
```

Comment:
Added the refactor here: 8dc13d74a5a2e360b21b76ce6c1bad13e6e76e19

The only "slowdown" should be in the case of when it *is* an object, in which case, it's two extra string comparisons.